monai/networks/nets/autoencoderkl.py

```diff
@@ -569,19 +441,45 @@ def __init__(
 
                 if reversed_attention_levels[i]:
                     blocks.append(
-                        _AttentionBlock(
+                        AttentionBlock(
                             spatial_dims=spatial_dims,
                             num_channels=block_in_ch,
                             norm_num_groups=norm_num_groups,
                             norm_eps=norm_eps,
-                            use_flash_attention=use_flash_attention,
                         )
                     )
 
             if not is_final_block:
-                blocks.append(
-                    _Upsample(spatial_dims=spatial_dims, in_channels=block_in_ch, use_convtranspose=use_convtranspose)
-                )
+                if use_convtranspose:
+                    blocks.append(
+                        Upsample(
+                            spatial_dims=spatial_dims, mode="deconv", in_channels=block_in_ch, out_channels=block_in_ch
+                        )
+                    )
+                else:
+                    post_conv = Convolution(
+                        spatial_dims=spatial_dims,
+                        in_channels=block_in_ch,
+                        out_channels=block_in_ch,
+                        strides=1,
+                        kernel_size=3,
+                        padding=1,
+                        conv_only=True,
+                    )
+                    blocks.append(
+                        Upsample(
+                            spatial_dims=spatial_dims,
+                            mode="nontrainable",
+                            in_channels=block_in_ch,
+                            out_channels=block_in_ch,
+                            interp_mode="nearest",
+                            scale_factor=2.0,
+                            post_conv=post_conv,
+                            align_corners=None,
+                        )
+                    )
+                    # rename postconv for compatibility with monai-generative
+                    blocks[-1].__dict__["_modules"]["conv"] = blocks[-1].__dict__["_modules"].pop("postconv")
```

Comment:
I think it's better to do the compatibility in the `load_old_state_dict`.  We can load the weight of the `conv` from old model to the `postconv` in the new model instead of modifying the module name. What do you think?
cc @ericspod